src/App.tsx

```diff
@@ -2,28 +1,54 @@
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42).map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [currentPage, setCurrentPage] = useState(1);
+  const [perPage, setPerPage] = useState(5);
+  const items = getNumbers(1, 42).map(n => `Item ${n}`);
+  const total = items.length;
+  const itemsPerPageOptions = [3, 5, 10, 20];
+
+  const handlePageChange = (page: number) => {
+    setCurrentPage(page);
+  };
+
+  const handlePerPageChange = (newPerPage: number) => {
+    setPerPage(newPerPage);
+    setCurrentPage(1);
+  };
+
+  const startIndex = Math.min((currentPage - 1) * perPage + 1, total);
+  const endIndex = Math.min(startIndex + perPage - 1, total);
+  const currentItems = items.slice(startIndex - 1, endIndex);
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        Page {currentPage} (items {startIndex} - {endIndex} of {total})
       </p>
 
       <div className="form-group row">
         <div className="col-3 col-sm-2 col-xl-1">
           <select
             data-cy="perPageSelector"
             id="perPageSelector"
-            className="form-control">
-            <option value="3">3</option>
-            <option value="5">5</option>
-            <option value="10">10</option>
-            <option value="20">20</option>
+            className="form-control"
+            value={perPage}
+            onChange={event => {
+              handlePerPageChange(+event.target.value);
```

Comment:
Casting the event target value to a number using the unary plus (+event.target.value) is clear, but consider using Number(event.target.value) for improved readability.